### PR TITLE
[Merged by Bors] - feat(Algebra/Order/Monoid/Unbundled/WithTop): add lemmas about `WithTop.map` and `WithBot.map`

### PR DESCRIPTION
**Pull request by Command-Master**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/WithBot.lean [hunk 1]
```diff
@@ -101,6 +101,14 @@ theorem map_coe (f : α → β) (a : α) : map f a = f a :=
 lemma map_eq_bot_iff {f : α → β} {a : WithBot α} :
     map f a = ⊥ ↔ a = ⊥ := Option.map_eq_none'
 
+theorem map_eq_some_iff {α β : Type*} {f : α → β} {y : β} {v : WithBot α} :
+    WithBot.map f v = .some y ↔ ∃ x, v = .some x ∧ f x = y := by
```
Comment (urkud) [inline Mathlib/Order/WithBot.lean:105]: Is it `Option.map_eq_some'`?

Comment (Command-Master) [reply Mathlib/Order/WithBot.lean]: Yes, updated

File: Mathlib/Algebra/Order/Monoid/Unbundled/WithTop.lean [hunk 2]
```diff
@@ -330,6 +338,24 @@ instance addMonoidWithOne : AddMonoidWithOne (WithTop α) :=
 @[simp] lemma top_ne_ofNat (n : ℕ) [n.AtLeastTwo] : (⊤ : WithTop α) ≠ no_index (OfNat.ofNat n) :=
   top_ne_natCast n
 
+@[simp] lemma map_ofNat {β : Type*} {f : α → β} (n : ℕ) [n.AtLeastTwo] :
+    WithTop.map f (no_index (OfNat.ofNat n : WithTop α)) = f n := map_coe f n
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Order/Monoid/Unbundled/WithTop.lean:342]: I think you want f (OfNat.ofNat n) on the rhs

File: Mathlib/Algebra/Order/Monoid/Unbundled/WithTop.lean [hunk 3]
```diff
@@ -73,6 +73,14 @@ theorem coe_lt_one [LT α] {a : α} : (a : WithTop α) < 1 ↔ a < 1 :=
 protected theorem map_one {β} (f : α → β) : (1 : WithTop α).map f = (f 1 : WithTop β) :=
   rfl
 
+@[to_additive]
+theorem map_eq_one_iff {α β : Type*} {f : α → β} {v : WithTop α} [One β] :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Order/Monoid/Unbundled/WithTop.lean:77]: Do you need to redeclare the types everywhere?

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**